x-pack/plugins/apm/server/routes/storage_explorer/get_storage_details_per_processor_event.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  termQuery,
+  kqlQuery,
+  rangeQuery,
+} from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { Setup } from '../../lib/helpers/setup_request';
+import {
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+  TIER,
+  INDEX,
+} from '../../../common/elasticsearch_fieldnames';
+import {
+  IndexLifecyclePhaseSelectOption,
+  indexLifeCyclePhaseToDataTier,
+} from '../../../common/storage_explorer_types';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { ApmPluginRequestHandlerContext } from '../typings';
+import {
+  getTotalIndicesStats,
+  getEstimatedSizeForDocumentsInIndex,
+} from './indices_stats_helpers';
+import { RandomSampler } from '../../lib/helpers/get_random_sampler';
+
+export async function getStorageDetailsPerProcessorEvent({
+  setup,
+  context,
+  indexLifecyclePhase,
+  randomSampler,
+  start,
+  end,
+  environment,
+  kuery,
+  serviceName,
+}: {
+  setup: Setup;
+  context: ApmPluginRequestHandlerContext;
+  indexLifecyclePhase: IndexLifecyclePhaseSelectOption;
+  randomSampler: RandomSampler;
+  start: number;
+  end: number;
+  environment: string;
+  kuery: string;
+  serviceName: string;
+}) {
+  const { apmEventClient } = setup;
```

Comment:
Side note:

@MiriamAparicio are you still looking to [splitting up `setup`](https://github.com/elastic/kibana/issues/113386)? I think that would be great. In most cases (like here) we just need `apmEventClient`. I just looked at how we consume `setup`:

```ts
const { apmEventClient } = setup; // 77 usages 
const { internalClient, indices } = setup; // 11 usages
const { apmEventClient, config } = setup; // 6 usages
const { ml } = setup; // 2 usages
const { ml, indices } = setup;  // 1 usage
const { indices } = setup; // 1 usage introduced by this PR
```